github/github.go

```diff
@@ -390,7 +390,7 @@ func parseRate(r *http.Response) Rate {
 // The provided ctx must be non-nil. If it is canceled or times out,
 // ctx.Err() will be returned.
 func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Response, error) {
-	req = req.WithContext(ctx)
+	ctx, req = addContext(ctx, req)
```

Comment:
Question about the path on App Engine. Again, I'm not familiar with how it works, but I'd like to understand for this PR.

Given that ctx is not really used, and `addContext` modifies only `ctx`, not `req`, does it actually have an effect on App Engine? Do you know how?